packages/gatsby/src/commands/serve.ts

```diff
@@ -295,8 +295,6 @@ module.exports = async (program: IServeProgram): Promise<void> => {
             }
             return res.send(results)
           }
-
-          return res.status(404).sendFile(`404.html`, { root })
```

Comment:
This was breaking some code paths in e2e/prod-runtime tests that were previously not exposed because engines were not generated (this PR adds first SSR pages, so this code path was finally hit in those tests). 

Later we have request handler for matchPaths (SSG matchpaths, not handled by engine) and sending 404 was just blocking it. We still have final fallback 404 later in final request handler.